### ISSUE TITLE
SLIM Loss Function

### ISSUE THREAD

**Issue by nasimsonboli** [guoguibing/librec#187]
Hi,
I have a question about the implementation of SLIM which is according to this paper, Ning, Xia, and George Karypis. "Slim: Sparse linear methods for top-n recommender systems.". 
In the loss function the error is multiplied by 1/2, but it is not the same in Librec's implementation. Is there a specific reason that I'm missing?

The implementation in Librec:
loss += errors + 0.5 * regL2Norm * coefficient * coefficient + regL1Norm * coefficient;
the suggestion:
loss += **0.5** * errors + 0.5 * regL2Norm * coefficient * coefficient + regL1Norm * coefficient;

Thanks,
Nasim
